### PR TITLE
Multipart form data

### PR DESCRIPTION
**Pull request by smolchanovsky**

### CODE REVIEW

File: tests/NClient/NClient.Tests/ClientTests/FormUrlencodedClientTest.cs [hunk 1]
```diff
@@ -14,54 +15,52 @@ namespace NClient.Tests.ClientTests
...
-        public async Task PostAsync_WithDictionary_ShouldReturnIntInBody()
+        public async Task PostAsync_WithDictionary_ShouldReturnOk()
         {
-            const int id = 1;
-            const int value = 1;
             var keyValues = new Dictionary<string, object>
             {
-                [nameof(id)] = id,
-                [nameof(value)] = value
+                ["id"] = 1,
```
Comment (Kingmidas74) [inline tests/NClient/NClient.Tests/ClientTests/FormUrlencodedClientTest.cs:38]: Why did you replace nameof() to strings?

File: src/NClient/NClient.Abstractions/Providers/Transport/Common/Content.cs [hunk 2]
```diff
@@ -22,7 +22,7 @@ public class Content : IContent
         public Content(Stream? streamContent = null, string? encoding = null, IMetadataContainer? headerContainer = null)
         {
             Stream = streamContent ?? new MemoryStream(Array.Empty<byte>());
-            Encoding = string.IsNullOrEmpty(encoding) ? Encoding.UTF8 : Encoding.GetEncoding(encoding);
+            Encoding = string.IsNullOrEmpty(encoding) ? null : Encoding.GetEncoding(encoding);
```
Comment (Kingmidas74) [inline src/NClient/NClient.Abstractions/Providers/Transport/Common/Content.cs:25]: Are you sure this changes won't cause NRE-like exception when request contains json-data and files and etc?

File: tests/NClient/NClient.Tests/ClientTests/FormUrlencodedClientTest.cs (hunk 1 above)
Comment (smolchanovsky) [reply tests/NClient/NClient.Tests/ClientTests/FormUrlencodedClientTest.cs:38]: It doesn't matter ) Keys don't depend on variable names

File: src/NClient/NClient.Abstractions/Providers/Transport/Common/Content.cs (hunk 2 above)
Comment (smolchanovsky) [reply src/NClient/NClient.Abstractions/Providers/Transport/Common/Content.cs:25]: Now we can use `Microsoft.AspNetCore.Http.IFormFile` interface to transport files. This interface doesn't have encoding, so I made `Encoding` field optional

File: tests/NClient/NClient.Tests/ClientTests/FormUrlencodedClientTest.cs (hunk 1 above)
Comment (Kingmidas74) [reply tests/NClient/NClient.Tests/ClientTests/FormUrlencodedClientTest.cs:38]: So... I asked about reasons of that changes. By my opinion, strings is harder for refactoring. (Yes, there are just two simple usage, but IT doesn't matter)

File: src/NClient/NClient.Abstractions/Providers/Transport/Common/Content.cs (hunk 2 above)
Comment (Kingmidas74) [reply src/NClient/NClient.Abstractions/Providers/Transport/Common/Content.cs:25]: And again.. It doesn't answered to my question. That about possible of NRE?

File: tests/NClient/NClient.Tests/ClientTests/FormUrlencodedClientTest.cs (hunk 1 above)
Comment (smolchanovsky) [reply tests/NClient/NClient.Tests/ClientTests/FormUrlencodedClientTest.cs:38]: So, if keys don't depend on variable names, why should I use `nameof`? ) Strictly speaking, `nameof` is little bit more complex than string, which means I made code simpler without losing quality

File: src/NClient/NClient.Abstractions/Providers/Transport/Common/Content.cs (hunk 2 above)
Comment (smolchanovsky) [reply src/NClient/NClient.Abstractions/Providers/Transport/Common/Content.cs:25]: I think you should formulate the questions more clearly. What exactly are you afraid of? That current code throw an NRE exception? Answer is no - we have tests. The future code may throw an NRE exception? Again, we have tests, in addition, we have a warnings to check null values

File: tests/NClient/NClient.Tests/ClientTests/FormUrlencodedClientTest.cs (hunk 1 above)
Comment (Kingmidas74) [reply tests/NClient/NClient.Tests/ClientTests/FormUrlencodedClientTest.cs:38]: Ok. I suggest you to use something one: nameof() or strings in every tests.

Comment (smolchanovsky) [reply tests/NClient/NClient.Tests/ClientTests/FormUrlencodedClientTest.cs]: Well, I've back `nameof` for uniformity

Comment (Kingmidas74) [reply tests/NClient/NClient.Tests/ClientTests/FormUrlencodedClientTest.cs]: Good job!

**Review by Kingmidas74 — APPROVED** [review]

**Pull request merged**